### PR TITLE
Target the Mamba in_proj in the Nemotron 3 LoRA example

### PR DESCRIPTION
**Pull request by behroozazarkhalili**
The Nemotron example omitted the Mamba `in_proj` matrices from its LoRA targets, so they never trained. I added that target to the script and notebook, and a CPU backward test confirmed a nonzero gradient in its LoRA matrix. This PR stacks on #6866, which removes the dead target from the same line.
Follow-up to #6773.

---

> [!NOTE]
> **High Risk**
> Strips the experimental PPO trainer, examples, and tests (breaking for PPO users) alongside broad doc and agent-tooling changes; the Nemotron LoRA tweak itself is low risk.
> 
> **Overview**
> Adds **`in_proj`** to the Nemotron 3 SFT LoRA target list in `sft_nemotron_3.py` and the matching notebook so Mamba input projections are adapted alongside attention and MLP modules.
> 
> Introduces a dedicated **Training Beyond 1M Tokens** guide (`long_context_training.md`), an **8-GPU context-parallel Qwen3-8B example** (`examples/sft_qwen3_8b_1m_context/`), and updates the docs nav and “What’s New” to feature long-context training. The long sequence-parallelism section is **removed from** `distributing_training.md` in favor of linking to the new guide.
> 
> **Removes experimental PPO** from the public surface: PPO trainer docs, examples (`ppo_sentiment`, `ppo_tldr`), paper-index entry, PEFT integration section, and the `trl.experimental.ppo` package plus its large test suite. Reference-model tests move to `tests/experimental/test_utils.py` using plain `AutoModelForCausalLM`.
> 
> Contributor/agent workflow updates: tracked **`.agents/skills/`** (including `update-paper-index`), **`make codex` removed**, Claude symlinks to `.agents/skills`, and **grep/diff guidance** in `.ai/AGENTS.md` for keeping duplicated trainer blocks aligned.
> 
> Smaller fixes: BCO doc snippet typos, expanded **HF Jobs** Docker/`hf jobs run` docs, **Kernels Hub** version pinning notes, trainer taxonomy on the docs index, and **bf16 hidden / fp32 `lm_head`** tests for chunked SFT and distillation losses.
> 
> <sup>Reviewed by [Cursor Bugbot](https://cursor.com/bugbot) for commit a2073962062080b407d543f595c0f043019fca77. Bugbot is set up for automated code reviews on this repo. Configure [here](https://www.cursor.com/dashboard/bugbot).</sup>

### CODE REVIEW

**Comment by behroozazarkhalili** [comment]
Restacked on the refreshed `#6866` branch, which moved to the new example path after the `#6820` reorganization.

`examples/scripts/sft_nemotron_3.py` is now `examples/sft_nemotron_3/sft_nemotron_3.py`. Merging the updated base produced no conflict, and this PR's own delta is unchanged: it adds `in_proj` to the LoRA targets on top of `#6866` removing `gate_proj`. The resulting line is:

```
--lora_target_modules q_proj k_proj v_proj o_proj up_proj down_proj in_proj
```

`ruff check` and `ruff format --check` at the CI-pinned 0.13.3 both pass. As on `#6866`, the pinned doc-builder check fails on this file, and it fails identically on unmodified `main`, so it is pre-existing rather than introduced here.

This PR still targets `fix/6773-nemotron-lora-target-modules` rather than `main`, so it should land after `#6866`.

File: trl/trainer/sft_trainer.py [hunk 1]
```diff
@@ -1168,7 +1168,10 @@ def __init__(
         # BFD packing requires padding-free mode; otherwise, the collator outputs padded attention masks, causing
         # FlashAttention to ignore position_ids and recompute them incorrectly from the padded attention mask.
         self.padding_free = args.padding_free or (args.packing and args.packing_strategy in {"bfd", "bfd_split"})
-        use_flash_attention = model.config._attn_implementation in FLASH_ATTENTION_VARIANTS
+        # A hub kernel can be requested with a revision and/or a kernel name (`repo_id@revision:kernel_name`), while
+        # the variants above are bare repo ids, so compare against the repo id alone.
+        attn_implementation = model.config._attn_implementation.split("@")[0].split(":")[0]
+        use_flash_attention = attn_implementation in FLASH_ATTENTION_VARIANTS
```
[the remark this answers is not in the document]
Comment (behroozazarkhalili) [reply trl/trainer/sft_trainer.py:1178]: This line is not part of this PR. It comes from #6985 on `main` (`sft_trainer.py:1173`, merged 2026-09-02) and reached this branch through the merge-forward; the diff here is two example files. On transformers 5.11.0 the attribute is `'sdpa'` after `from_pretrained`, `from_config`, and the direct constructor, so I could not reach the `None` case either. If it is reachable, it belongs against #6985 or as its own issue.

File: trl/trainer/sft_trainer.py [hunk 2]
```diff
@@ -100,7 +100,10 @@ class _ChunkedCELMHeadOutput(CausalLMOutputWithPast):
 
 def _chunk(h, w, b, lbl, logit_scale, final_logit_softcapping):
     with maybe_gather_lm_head_ctx(w, b):
-        logits = h.float() @ w.float().t()
+        # Project in the compute dtype and upcast only for the softmax, like `"nll"` and
+        # `transformers`' own `ForCausalLMLoss` do. Matching the weight to the hidden-states dtype keeps the
+        # matmul in that dtype, which is what `lm_head`'s own forward computes in.
+        logits = (h @ w.to(h.dtype).t()).float()
```
[the remark this answers is not in the document]
Comment (behroozazarkhalili) [reply trl/trainer/sft_trainer.py:106]: This line is not part of this PR. It arrived through the merge of `main` and comes from #6863 (fb62e4b4, "Do the chunked-CE `lm_head` projection on tensor cores"); this branch's diff against `main` for `trl/trainer/sft_trainer.py` is empty. If the per-chunk cast is worth hoisting, that belongs in a follow-up to #6863, not here.
